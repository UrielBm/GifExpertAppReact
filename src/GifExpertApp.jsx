import React, { Fragment,useState } from 'react'
import { AddCategorias } from './Components/AddCategorias';
import { GifGrid } from './Components/GifGrid';

export const GifExpertApp = () => {
   //const categorias =['Perros','Anime','Videojuegos', 'One-punch'];
   const [categorias, setCategorias] = useState(['Perros']);
//    const hadleAddCategorias=()=>{
//         // setCategorias([...categorias,'Pokemon']);
//         setCategorias(categ=>[...categ,'Pokemon']);
//    };

    return (
        <Fragment>
            <div className="wrapper-main">
            <h2 className="Title">**GifExpertApp**</h2>
            <AddCategorias setCategorias={setCategorias} />
            <hr className="Linea"/>
            <ol>
            {
                categorias.map(category=>(
                     <GifGrid key={category} categoria={category} />
                ))
            }
            </ol>
            <footer className="footer">Aplicación creada por: Benítez Medina Uriel</footer>
            </div>
        </Fragment>
    )
}
