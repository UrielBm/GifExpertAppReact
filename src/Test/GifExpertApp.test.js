import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('realizando pruebas en el componente <GifExpertApp />', () => {
    test('Muestra de manwera corrrecta el componente <GifExpertApp  ', () => {
        const wrapper=shallow(<GifExpertApp />);
        expect(wrapper).toMatchSnapshot();
    });
    test('debe de mostrar una lista de categorias', () => {
        const categorias= ['Goku', 'Pokemon'];
        const wrapper=shallow(<GifExpertApp defaultCategorias={categorias} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);
    })
    
    
})
