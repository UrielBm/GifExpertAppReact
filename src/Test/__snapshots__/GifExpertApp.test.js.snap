// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`realizando pruebas en el componente <GifExpertApp /> Muestra de manwera corrrecta el componente <GifExpertApp   1`] = `
<Fragment>
  <div
    className="wrapper-main"
  >
    <h2
      className="Title"
    >
      **GifExpertApp**
    </h2>
    <AddCategorias
      setCategorias={[Function]}
    />
    <hr
      className="Linea"
    />
    <ol />
    <footer
      className="footer"
    >
      Aplicación creada por: Benítez Medina Uriel
    </footer>
  </div>
</Fragment>
`;

exports[`realizando pruebas en el componente <GifExpertApp /> debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <div
    className="wrapper-main"
  >
    <h2
      className="Title"
    >
      **GifExpertApp**
    </h2>
    <AddCategorias
      setCategorias={[Function]}
    />
    <hr
      className="Linea"
    />
    <ol>
      <GifGrid
        categoria="Goku"
        key="Goku"
      />
      <GifGrid
        categoria="Pokemon"
        key="Pokemon"
      />
    </ol>
    <footer
      className="footer"
    >
      Aplicación creada por: Benítez Medina Uriel
    </footer>
  </div>
</Fragment>
`;
